### DISCUSSION TITLE
Can we navigate to a tiddler without changing the $:/HistoryList?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5796] [General]
I'm working on a mechanism that preserves the focused domNode over refresh cycles
There are also keyboard shortcuts to navigate from one tiddler in the StoryRiver to another
If there's a tiddler in edit-mode with the title focused and I navigate to another tiddler, the `$:/HistoryList` is changed which causes the page to refresh and the `edit-text` widgets of the tiddler in edit-mode also somehow do a full refresh.

Now I'm asking if it's possible to avoid that behavior either by not making the `edit-text` widgets do a full refresh when `$:/HistoryList` changes (???) or by somehow avoiding the change of `$:/HistoryList`

**Comment by BurningTreeC** [comment]
I'm not sure if the `edit-text` widgets do a full refresh, but they render again

**Reply by AnthonyMuscio** [reply]
BTC,

Perhaps this is off the mark, but just in case, as you may know, the history list tiddler can be set. See https://tiddlywiki.com/#tv-history-list%20Variable

I wonder if you change change this variable, or present an alternate histor tiddler if you could fool the mechaisium in some way?

Just a passing thought.
Tony

**Comment by pmario** [comment]
As written here. https://github.com/Jermolene/TiddlyWiki5/issues/4782#issuecomment-671028109 ... probably not, with the existing navigation handling. ... Which we didn't fix yet, since several different core elements are involved. ...

So IMO we need to improve the underlying mechanism first. 

If we introduce a hack for your "selection handling", imo it will make fixing the underlying problem harder. 

Just a thought.

**Comment by Jermolene** [comment]
> I'm working on a mechanism that preserves the focused domNode over refresh cycles
> 
> There are also keyboard shortcuts to navigate from one tiddler in the StoryRiver to another
> 
> If there's a tiddler in edit-mode with the title focused and I navigate to another tiddler, the `$:/HistoryList` is changed which causes the page to refresh and the `edit-text` widgets of the tiddler in edit-mode also somehow do a full refresh.
> 
> 
> 
> Now I'm asking if it's possible to avoid that behavior either by not making the `edit-text` widgets do a full refresh when `$:/HistoryList` changes (???) or by somehow avoiding the change of `$:/HistoryList`

The refresh mechanism of the edit text widgets avoids changing the value of the text area if it contains the focus. We do it because when a tiddler is being edited we can guarantee that the text in the control will always be up to date.

We could explore tweaking the logic. For example, we might check whether the text we've got from the store is different to the text in the control before we assign it.
